feat(logs): pass filters into attributes selector

frankh opened the pull request:
## Problem

currently adding filters always shows _all_ our attributes from all logs

## Changes

make it so service name filters and other attribute filters get passed into the attributes endpoint, meaning you only see relevant attributes when adding new ones which already exist

also updated the local log collector to enrich logs with resource attributes to make local testing better

https://github.com/user-attachments/assets/a9131fc0-519a-4914-8691-cd9a75c30d00


## How did you test this code?

added new test fixtures and test cases for logs query runner
<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?
yes
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->
<!-- Removing this section does not mean the changelog bot won't pick it up, because *some people* like to not use the template, so we can't rely on it existing. -->

frankh committed:
pass filters into log attributes query to narrow down choices

frankh committed:
add new hogql settings

frankh committed:
use property scope instead of attribute type to switch to log_attributes filter mode

frankh committed:
add proper logs clickhouse tests

frankh committed:
remove unused import

frankh committed:
handle bad serviceNames json

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

frankh committed:
fmt

jonmcwest commented on frontend/src/lib/components/PropertyFilters/types.ts:
```diff
@@ -61,4 +62,5 @@ export interface PropertyFilterInternalProps {
     exactMatchFeatureFlagCohortOperators?: boolean
     hideBehavioralCohorts?: boolean
     addFilterDocLink?: string
+    endpointFilters?: Record<string, any>
```
Would be good to have a type for this record's values if possible instead of `any` - I might be missing something though.

frankh committed:
refactor log query runner inheritance and fix type errors

frankh committed:
fix type issues

frankh committed:
filter values too

frankh committed:
missing type

frankh committed:
add attributes and values unit tests

frankh committed:
# This is a combination of 5 commits.
# This is the 1st commit message:

fix type issues

# The commit message #2 will be skipped:

# fix test types/response

# The commit message #3 will be skipped:

# null not undefined

# The commit message #4 will be skipped:

# wip

# The commit message #5 will be skipped:

# wip

frankh committed:
fix type issues
